### PR TITLE
Add RSS feed 

### LINKED ISSUE #11: Add RSS feed for the blog articles
I really like your blog articles, would you accept a pull request, that adds an RSS feed to your blog?

### PR DESCRIPTION
**Pull request by dweidner**
I basically used the sample provided by the [`eleventy-plugin-rss`](https://github.com/11ty/eleventy-plugin-rss) plugin and tried to adapt it to your coding style. I had fun experimenting with the JavaScript template which I haven’t used before in the 11ty context.

The [`eleventy-plugin-rss`](https://github.com/11ty/eleventy-plugin-rss) plugin provides an additional function named [`htmlToAbsoluteUrls`](https://github.com/11ty/eleventy-plugin-rss/blob/master/src/htmlToAbsoluteUrls.js) which does what its name suggests. Unfortunately it runs [`posthtml`](https://github.com/posthtml/posthtml) asynchronously in the background which increases the complexity. So far you don't use internal links in the excerpt. Do you think it is necessary to run the post processor?

Fixes #11

### CODE REVIEW

**Comment by dweidner** [comment]
You could probably do completely without the RSS plugin as a dependency if you implement your own helper functions for the ISO dates/absolute URLs.

File: src/rss.11ty.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+              <content type="html">${postExcerpt}</content>
+            </entry>
+          `;
+        },
+        ''
+      )}
+    </feed>
+  `;
+
+  return feed.trim();
```
Comment (dweidner) [inline src/rss.11ty.js:70]: This is necessary to have valid XML output. No spaces allowed in front of the xml declaration.

File: src/rss.11ty.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  return {
+    permalink: 'feed.xml',
+    site: {
+      title: 'Oliver Phillips - Frontend Engineer',
+      subtitle: 'I design and develop user experiences.',
+      url: 'https://oliverjam.es/',
+      author: {
+        name: 'Oliver Phillips',
+        email: 'hello@oliverjam.es'
+      },
```
Comment (oliverjam) [inline src/rss.11ty.js:16]: I should probably move this stuff into a global data file so I can edit it in one place. Not a blocker for this PR though, I'll get round to it at some point

**Review by oliverjam — APPROVED** [review]
This is incredible–thank you for taking the time and effort to not only implement but implement in the style of the existing codebase

**Comment by oliverjam** [comment]
Does the RSS plugin only add the filters? If so I will probably drop the dependency later, but for now if this works I would rather have a a working RSS sooner 😄

**Comment by dweidner** [comment]
Yes, the plugin basically provides four filters for Nunjuck templates to make your life easier when generating a feed. There is also an example of what the Nunjuck template for a feed might look like and how to apply the provided filters. 

Unfortunately these filters are registered as Nunjuck's filters and cannot be used within JavaScript Templates directly. Therefore, I had to load them from the source directory of the plugin instead.

If, as in my implementation, you display excerpts in your feed instead of the complete content, I think it is also reasonable to remove HTML tags completely. This way URLs don't have to be rewritten (by using `htmlToAbsoluteUrls` or a custom filter).

**Comment by oliverjam** [comment]
Yeah I try to avoid any HTML in my excerpts, since I use them in page meta descriptions too. I think I currently have a single link inside one that I need to get rid of.

Thank you again for your work on this!

**Comment by dweidner** [comment]
You’re welcome. Keep up the good work. Looking forward to your next article.

**Pull request merged**